src/Components/ServiceScene/Breakdowns/Patterns/PatternNameLabel.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useRef, useState } from 'react';
+import { AdHocFiltersVariable, sceneGraph } from '@grafana/scenes';
+import { Spinner, Toggletip } from '@grafana/ui';
+import { getLokiDatasource } from 'services/scenes';
+import { IndexScene } from 'Components/IndexScene/IndexScene';
+import { VAR_FILTERS } from 'services/variables';
+import { buildLokiQuery } from 'services/query';
+import { PatternFieldLabelStats } from './PatternFieldLabelStats';
+import { LogLabelStatsModel, TimeRange } from '@grafana/data';
+import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from 'services/analytics';
+
+interface PatternNameLabelProps {
+  exploration: IndexScene;
+  pattern: string;
+}
+
+const LINE_LIMIT = 2000;
+
+export const PatternNameLabel = ({ exploration, pattern }: PatternNameLabelProps) => {
+  const patternIndices = extractPatternIndices(pattern);
+  const [stats, setStats] = useState<LogLabelStatsModel[][] | undefined>(undefined);
+
+  // Refs to store the previous values of query and timeRange
+  const previousQueryRef = useRef<string>('');
+  const previousTimeRangeRef = useRef<TimeRange>();
+
+  const handleClick = async () => {
+    reportAppInteraction(USER_EVENTS_PAGES.service_details, USER_EVENTS_ACTIONS.service_details.pattern_field_clicked);
+    const query = constructQuery(
+      pattern,
+      patternIndices,
+      sceneGraph.lookupVariable(VAR_FILTERS, exploration) as AdHocFiltersVariable
+    );
+    const datasource = await getLokiDatasource(exploration);
+    const currentTimeRange = sceneGraph.getTimeRange(exploration).state.value;
+
+    // If the query and timeRange are the same as the previous ones, do not re-query
+    if (stats && query === previousQueryRef.current && currentTimeRange === previousTimeRangeRef.current) {
+      return;
+    }
+
+    // Update the refs with the new values
+    previousQueryRef.current = query;
+    previousTimeRangeRef.current = currentTimeRange;
+
+    datasource
```

Comment:
Need to catch errors and display error state. Currently failures load forever.
<img width="1728" alt="image" src="https://github.com/grafana/explore-logs/assets/109082771/132498c0-24f7-43cd-a954-f8c3c9e24ee9">

Reply:
Request that's failing from above:

Expr:
```
{service_name=`grafana/grafana`} |> `logger=sqlstore.metrics traceID=<_> t=<_>:<_>:<_>.<_> level=debug msg="query finished" status=success elapsedtime=<_>.<_> sql="SELECT `id`, `org_id`, `title`, `condition`, `data`, `updated`, `interval_seconds`, `version`, `uid`, `namespace_uid`, `dashboard_uid`, `panel_id`, `rule_group`, `rule_group_idx`, `record`, `no_data_state`, `exec_err_state`, `for`, `annotations`, `labels`, `is_paused`, `notification_settings` FROM `alert_rule` WHERE `rule_group` IN (?)" error=null` | pattern `logger=sqlstore.metrics traceID=<field_1> t=<field_2>:<field_3>:<field_4>.<field_5> level=debug msg="query finished" status=success elapsedtime=<field_6>.<field_7> sql="SELECT `id`, `org_id`, `title`, `condition`, `data`, `updated`, `interval_seconds`, `version`, `uid`, `namespace_uid`, `dashboard_uid`, `panel_id`, `rule_group`, `rule_group_idx`, `record`, `no_data_state`, `exec_err_state`, `for`, `annotations`, `labels`, `is_paused`, `notification_settings` FROM `alert_rule` WHERE `rule_group` IN (?)" error=null` | keep field_1 ,field_2 ,field_3 ,field_4 ,field_5 ,field_6 ,field_7 | line_format ""
```

Full request:
```
ds_type: loki
requestId: 1_1
{"queries":[{"refId":"A","queryType":"range","editorMode":"code","supportingQueryType":"grafana-lokiexplore-app","maxLines":2000,"expr":"{service_name=`grafana/grafana`} |> `logger=sqlstore.metrics traceID=<_> t=<_>:<_>:<_>.<_> level=debug msg=\"query finished\" status=success elapsedtime=<_>.<_> sql=\"SELECT `id`, `org_id`, `title`, `condition`, `data`, `updated`, `interval_seconds`, `version`, `uid`, `namespace_uid`, `dashboard_uid`, `panel_id`, `rule_group`, `rule_group_idx`, `record`, `no_data_state`, `exec_err_state`, `for`, `annotations`, `labels`, `is_paused`, `notification_settings` FROM `alert_rule` WHERE `rule_group` IN (?)\" error=null` | pattern `logger=sqlstore.metrics traceID=<field_1> t=<field_2>:<field_3>:<field_4>.<field_5> level=debug msg=\"query finished\" status=success elapsedtime=<field_6>.<field_7> sql=\"SELECT `id`, `org_id`, `title`, `condition`, `data`, `updated`, `interval_seconds`, `version`, `uid`, `namespace_uid`, `dashboard_uid`, `panel_id`, `rule_group`, `rule_group_idx`, `record`, `no_data_state`, `exec_err_state`, `for`, `annotations`, `labels`, `is_paused`, `notification_settings` FROM `alert_rule` WHERE `rule_group` IN (?)\" error=null` | keep field_1 ,field_2 ,field_3 ,field_4 ,field_5 ,field_6 ,field_7 | line_format \"\"","legendFormat":"","datasource":{"type":"loki","uid":"e80c5725-e146-4f8b-bdf7-82a037b21cbc"},"datasourceId":350,"intervalMs":0}],"from":"1719415107085","to":"1719416007086"}
```

Pattern:
```
logger=sqlstore.metrics traceID=<_> t=<_>:<_>:<_>.<_> level=debug msg="query finished" status=success elapsedtime=<_>.<_> sql="SELECT `id`, `org_id`, `title`, `condition`, `data`, `updated`, `interval_seconds`, `version`, `uid`, `namespace_uid`, `dashboard_uid`, `panel_id`, `rule_group`, `rule_group_idx`, `record`, `no_data_state`, `exec_err_state`, `for`, `annotations`, `labels`, `is_paused`, `notification_settings` FROM `alert_rule` WHERE `rule_group` IN (?)" error=null
```

Response:
```
{
    "results": {
        "A": {
            "error": "parse error at line 1, col 173: syntax error: unexpected IDENTIFIER",
            "errorSource": "downstream",
            "status": 500
        }
    }
}
```

Found in Grafana logging dev

Reply:
Errors are now being handled. I will adjust styles in general in further commits.

![imagen](https://github.com/grafana/explore-logs/assets/1069378/6cbd7eb7-e8d1-4b6c-916d-b807c35ff766)